src/window.cpp

```diff
@@ -531,21 +525,23 @@ void Window::OnFocusLost(bool)
  */
 void Window::RaiseButtons(bool autoraise)
 {
-	for (uint i = 0; i < this->nested_array_size; i++) {
-		if (this->nested_array[i] == nullptr) continue;
-		WidgetType type = this->nested_array[i]->type;
+	for (auto &pair : this->widget_lookup) {
+		WidgetType type = pair.second->type;
+		NWidgetCore *wid = dynamic_cast<NWidgetCore *>(pair.second);
 		if (((type & ~WWB_PUSHBUTTON) < WWT_LAST || type == NWID_PUSHBUTTON_DROPDOWN) &&
-				(!autoraise || (type & WWB_PUSHBUTTON) || type == WWT_EDITBOX) && this->IsWidgetLowered(i)) {
-			this->RaiseWidget(i);
-			this->SetWidgetDirty(i);
+				(!autoraise || (type & WWB_PUSHBUTTON) || type == WWT_EDITBOX) && wid->IsLowered()) {
+			wid->SetLowered(false);
+			wid->SetDirty(this);
```

Comment:
This change feels unexpected. Was it intentional? And if so, is it really part of this PR?

Reply:
Kinda. The methods on Windows basically just use GetWidget<>(i), and call the method on the widget... because we iterate widget_lookup, we already have the widget, so can just call them directly.

But I could undo it and replace i with pair.first instead.

Reply:
The raised vs lowered was making me go: euhm?